src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiManifest.groovy

```diff
@@ -141,6 +146,44 @@ class JpiManifest extends Manifest {
     }
 
     private static listUpDependencies(Configuration c, boolean optional, StringBuilder buf) {
+        // The category attribute has been introduced in Gradle '5.3'
+        if (GradleVersion.current() >= GradleVersion.version('5.3')) {
+            listUpResolvedDependencies(c, optional, buf)
+        } else {
+            legacyListUpDependencies(c, optional, buf)
+        }
+    }
+
+    private static listUpResolvedDependencies(Configuration c, boolean optional, StringBuilder buf) {
+        def categoryAttribute = Attribute.of(Category.CATEGORY_ATTRIBUTE.name, String)
+
+        c.incoming.resolutionResult.root.dependencies.each { DependencyResult result ->
```

Comment:
@jjohannes, can you take a look if I got this right?

Reply:
Added a review 👇